app/src/main/java/net/squanchy/schedule/domain/view/Schedule.kt

```diff
@@ -1,12 +1,50 @@
 package net.squanchy.schedule.domain.view
 
-data class Schedule(val pages: List<SchedulePage>) {
+import net.squanchy.support.system.CurrentTime
+import org.joda.time.DateTimeZone
+
+private const val NOT_FOUND_INDEX = -1
+private const val FIRST_PAGE_INDEX = 0
+
+private const val CURRENT_SLOT_THRESHOLD = .6f
+
+data class Schedule(val pages: List<SchedulePage>, val timezone: DateTimeZone) {
 
     val isEmpty: Boolean
         get() = pages.isEmpty()
 
     companion object {
 
-        fun create(pages: List<SchedulePage>) = Schedule(pages)
+        fun create(pages: List<SchedulePage>, timezone: DateTimeZone) = Schedule(pages, timezone)
     }
+
+    fun findTodayIndexOrDefault(currentTime: CurrentTime) =
+        pages
+            .indexOfFirst { page ->
+                val now = currentTime.currentDateTime().toDateTime(timezone)
+                page.date.toLocalDate().isEqual(now.toLocalDate())
+            }
+            .let {
+                when (it) {
+                    NOT_FOUND_INDEX -> FIRST_PAGE_INDEX
+                    else -> it
+                }
+            }
+
+    fun findNextEventForPage(page: SchedulePage, currentTime: CurrentTime) =
+        page.events
+            .firstOrNull { event ->
+                val startDateTime = event.startTime.toDateTime().withZone(event.timeZone)
+                val currentDateTime = currentTime.currentDateTime().toDateTime().withZone(event.timeZone)
```

Comment:
and here, extracting the current time inside the method but outside the lambda

Reply:
I can't really do it here, since we need the timezone for the current event. We could assume equal timezones across events, but that conflicts with the data model (having a timezone as part of each `Event`).

Reply:
ah yeah makes sense